### ISSUE TITLE
ios 18 version get error

### ISSUE THREAD

**Issue by arifxpartbd** [pichillilorenzo/flutter_inappwebview#2279]
khanarif@Khans-MacBook-Pro gpspro % flutter build ios
Building com.gomax.gpspro for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: Y59JP677FD
Running pod install...                                             21.0s
Running Xcode build...                                                  
Xcode build done.                                           51.2s
Failed to build iOS app
Swift Compiler Error (Xcode): Method does not override any method from its superclass
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1431:25

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1035:20

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1038:16

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1047:20

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1050:16

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1327:25

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1912:8

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:2859:22

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:2870:22

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:2964:17

Encountered error while building for device.
khanarif@Khans-MacBook-Pro gpspro %

**Comment by kingace2056** [comment]
i've been facing same issue after upgrading to MacOS sequoia and upgrading Xcode.  Please if anyone knows the solution please let me know.

**Comment by SarfarazAhmed008** [comment]
Experiencing the same issue after upgrading to MacOS sequoia and upgrading Xcode 16.0.

**Comment by docaohuynh** [comment]
I have same error
```
Could not build the precompiled application for the device.
Swift Compiler Error (Xcode): Method does not override any method from its superclass
/Users/docaohuynh/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift:1431:25


Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/docaohuynh/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift:2859:22


Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/docaohuynh/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift:2870:22
```

**Comment by devplanet-dp** [comment]
Same here

**Comment by OliverNarramore** [comment]
Same here

**Comment by LeoAiolia** [comment]
Xcode16.0 has error，when debug running

**Comment by IncM** [comment]
For fast fix you can replace in **1432** line **InAppWebView** (flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift)
from
```swift
    public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: ((Any?, Error?) -> Void)? = nil) {
```
to
```swift
    public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: (@MainActor @Sendable (Any?, (any Error)?) -> Void)? = nil) {
```

**Comment by ke112** [comment]
I put this modified method, written a simple script, you can try to solve this problem   https://github.com/ke112/fixios18

**Comment by arifxpartbd** [comment]
> ```swift
>     public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: (@MainActor @Sendable (Any?, (any Error)?) -> Void)? = nil) {
> ```

Yes this is working, Thank you

**Comment by hleanghuor** [comment]
> For fast fix you can replace in **1432** line **InAppWebView** (flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift) from
> 
> ```swift
>     public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: ((Any?, Error?) -> Void)? = nil) {
> ```
> 
> to
> 
> ```swift
>     public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: (@MainActor @Sendable (Any?, (any Error)?) -> Void)? = nil) {
> ```

This working, Thanks you.

**Comment by paganellif** [comment]
Duplicate of #2278

**Comment by ToanDevMobile** [comment]
> For fast fix you can replace in **1432** line **InAppWebView** (flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.swift) from
> 
> ```swift
>     public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: ((Any?, Error?) -> Void)? = nil) {
> ```
> 
> to
> 
> ```swift
>     public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: (@MainActor @Sendable (Any?, (any Error)?) -> Void)? = nil) {
> ```

love you. thank you.

**Comment by JgomesAT** [comment]
I have the same eror, any finala solution?

the proposal solution is very dangerous, becous we are editing thje labray insie in .pubcache  and this is alwasy rewrite with pub get o whn you do a flutter clean

**Comment by IncM** [comment]
> I have the same error, any final solution?
> 
> the proposal solution is very dangerous, because we are editing the library inside in .pubcache and this is always rewrite with pub get o when you do a flutter clean

For temporary solution you can use my form with fix this problem. 
Add 
```yaml
dependency_overrides:
  # TODO: Recheck once flutter_inappwebview version >6.0.0 is released
  flutter_inappwebview_ios:
    git:
      url: https://github.com/IncM/flutter_inappwebview
      path: flutter_inappwebview_ios
      ref: cbc214c7b2cf5fd7996ff2e9e25d203946b74bc3
```
in your pubspec.yaml

**Comment by Michi-Vin** [comment]
Same issue. Updated XCode and game over :D

**Comment by 1057437122** [comment]
> i've been facing same issue after upgrading to MacOS sequoia and upgrading Xcode. Please if anyone knows the solution please let me know.

Yesterday I upgraded my Mac and Xcode, same issue appeared...

**Comment by arrrrny** [comment]
You can use my new package [zikzak_inappwebview](https://pub.dev/packages/zikzak_inappwebview) fork of this package that I will keep maintaining. Thanks @pichillilorenzo for developing/maintaining this package for a long time. I guess he is busy or on vacataion. I am the builder of Flutter Vendure (open source headless commerce framework) package. [Vendure](https://vendure.io/) Please open your PRs [here](https://github.com/arrrrny/zikzak_inappwebview) and I am happy to check them and merge them by Monday.

**Comment by muxtorovinomjon0227** [comment]
public override func evaluateJavaScript(_ javaScriptString: String, completionHandler: (@MainActor @Sendable (Any?, (any Error)?) -> Void)? = nil) {
    
    It worked for me. Thank you!

**Comment by HarshuPatel007** [comment]
I encountered the same error after updating Xcode.

Swift Compiler Error (Xcode): Method does not override any method from its superclass
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1431:25

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1035:20

Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)'
/Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView.
swift:1038:16

Please Let me know if you find any solution.

**Comment by jaxnz** [comment]
So it looks like this package has been forked with a fix, with on-going maintenance. It has fixed the issue for me.

https://pub.dev/packages/zikzak_inappwebview

**Comment by nemscep** [comment]
Perhaps easier solution is to install previous version of xcode 15.4 using [xcodes](https://formulae.brew.sh/cask/xcodes).
After installing, you just setup command line tools version in xcode locations settings to 15.4 instead of 16 and voila, everything is back to working 😄

**Comment by HarshuPatel007** [comment]
> I encountered the same error after updating Xcode.
> 
> Swift Compiler Error (Xcode): Method does not override any method from its superclass /Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView. swift:1431:25
> 
> Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)' /Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView. swift:1035:20
> 
> Swift Compiler Error (Xcode): Ambiguous use of 'evaluateJavaScript(_:completionHandler:)' /Users/khanarif/.pub-cache/hosted/pub.dev/flutter_inappwebview_ios-1.0.13/ios/Classes/InAppWebView/InAppWebView. swift:1038:16
> 

If you encounter the same error, you can find your solution by visiting this link.
https://github.com/pichillilorenzo/flutter_inappwebview/issues/2283#issuecomment-2367389122

**Comment by pichillilorenzo** [comment]
Released new version 6.1.0 with the fix, thanks.

**Comment by felipecastrosales** [comment]
Hello everyone! I opened an issue #2327 related to a new problem like this. I ask that you consider upvoting or interacting so that the maintainers can see this as soon as possible. Remember that these problems can happen to you - as is happening to me. :)
